spec/requests/api/snapshots_spec.rb

```diff
@@ -0,0 +1,149 @@
+        api_basic_authorize
+        vm = FactoryGirl.create(:vm_vmware)
+        FactoryGirl.create(:snapshot, :vm_or_template => vm)
+
+        run_get("#{vms_url(vm.id)}/snapshots")
+
+        expect(response).to have_http_status(:forbidden)
+      end
+    end
+  end
+
+  describe "GET /api/vms/:c_id/snapshots/:s_id" do
+    it "can show a VM's snapshot" do
+      api_basic_authorize(subcollection_action_identifier(:vms, :snapshots, :read, :get))
+      vm = FactoryGirl.create(:vm_vmware)
+      create_time = Time.zone.parse("2017-01-11T00:00:00Z")
+      snapshot = FactoryGirl.create(:snapshot, :vm_or_template => vm, :create_time => create_time)
+
+      run_get("#{vms_url(vm.id)}/snapshots/#{snapshot.id}")
+
+      expected = {
+        "create_time" => create_time.iso8601,
+        "href" => a_string_matching("#{vms_url(vm.id)}/snapshots/#{snapshot.id}"),
+        "id" => snapshot.id,
+        "vm_or_template_id" => vm.id
+      }
+      expect(response.parsed_body).to include(expected)
+      expect(response).to have_http_status(:ok)
+    end
+
+    it "will not show a snapshot unless authorized" do
+      api_basic_authorize
+      vm = FactoryGirl.create(:vm_vmware)
+      snapshot = FactoryGirl.create(:snapshot, :vm_or_template => vm)
+
+      run_get("#{vms_url(vm.id)}/snapshots/#{snapshot.id}")
+
+      expect(response).to have_http_status(:forbidden)
+    end
+
+    describe "POST /api/vms/:c_id/snapshots" do
+      it "can create a snapshot" do
+        api_basic_authorize(subcollection_action_identifier(:vms, :snapshots, :create))
+        vm = FactoryGirl.create(:vm_vmware)
+
+        run_post("#{vms_url(vm.id)}/snapshots", :description => "Alice's snapshot")
+
+        expected = {
+          "results" => [
+            a_hash_including(
+              "description"       => "Alice's snapshot",
+              "vm_or_template_id" => vm.id
+            )
+          ]
+        }
+        expect(response.parsed_body).to include(expected)
+        expect(response).to have_http_status(:ok)
+      end
+
+      it "will not create a snapshot unless authorized" do
+        api_basic_authorize
+        vm = FactoryGirl.create(:vm_vmware)
+
+        run_post("#{vms_url(vm.id)}/snapshots", :description => "Alice's snapshot")
+
+        expect(response).to have_http_status(:forbidden)
+      end
+    end
+
+    describe "POST /api/vms/:c_id/snapshots/:s_id" do
+      it "can update a snapshot" do
+        api_basic_authorize(action_identifier(:snapshots, :edit, :subresource_actions))
+        vm = FactoryGirl.create(:vm_vmware)
+        snapshot = FactoryGirl.create(:snapshot, :vm_or_template => vm, :description => "Alice's snapshot")
+
+        run_post("#{vms_url(vm.id)}/snapshots/#{snapshot.id}", :action => "edit", :description => "Bob's snapshot")
+
+        expected = {
+          "id"                => snapshot.id,
+          "href"              => a_string_matching(snapshots_url(snapshot.id)), # uh-oh....
```

Comment:
@abellotti bit confused by this behavior. "edit" will return `/api/snapshots/:id` (which does not exist), while "create" will return `/api/vms/:vm_id/snapshots/:id`. Any thoughts?